packages/react-components/src/components/RichTextEditor/Plugins/UpdateContentPlugin.tsx

```diff
@@ -70,22 +70,13 @@ export class UpdateContentPlugin implements EditorPlugin {
 
       case PluginEventType.ContentChanged:
         if (
-          event.source.toLowerCase() === 'cut' ||
-          (event.source.toLowerCase() === 'keyboard' && (event.data === Keys.BACKSPACE || event.data === Keys.DELETE))
+          event.source === ChangeSource.Cut ||
+          event.source === ChangeSource.Paste ||
```

Comment:
Why we need a paste event here if we already handle it in a separate plugin? (and call`onInsertInlineImage` callback there)

Reply:
We should do this in only 1 place, because currently we try to calculate added inline images in UpdatePlugin that will cause the call on insert callback but at the same time we try to call insert callback in oOpyPastePlugin. This will likely result in race conditions

Reply:
We need to add the paste here for an edge case: When user select an existing image, then paste in a image to replace the selected one, we only get ChangeSouce.Paste, so we loss track of the image that's removed.

Reply:
> We should do this in only 1 place, because currently we try to calculate added inline images in UpdatePlugin that will cause the call on insert callback but at the same time we try to call insert callback in CopyPastePlugin. This will likely result in race conditions

In UpdatePlugin and RichTextEditor, we are not calling the onInsertInlineImages. It is only called in the CopyPastePlugin. And CopyPastePlugin is triggered before UpdateContentPlugin.